File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AXNPC2.java
package Auxiliary_Modules;

public class AXNPC2 extends AXNPC{
    public AnnoyedQue annoyedQue = new AnnoyedQue(5);
    public AXNPC2(int replyStockLim, int outputChance) {
        super(replyStockLim, outputChance);
    }
    public void strLearn(String ear){
        // learns inputs containing strings that are repeatedly used by others
        annoyedQue.learn(ear);
        if (annoyedQue.isAnnoyed(ear)){
            this.responder.add(ear);
        }
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AXNPC2.kt
package auxiliary_modules

class AXNPC2(replyStockLim: Int, outputChance: Int) : AXNPC(replyStockLim, outputChance) {
    var annoyedQue = AnnoyedQue(5)
    fun strLearn(ear: String?) {
        // learns inputs containing strings that are repeatedly used by others
        annoyedQue.learn(ear!!)
        if (annoyedQue.isAnnoyed(ear)) {
            responder.add(ear)
        }
    }
}